### PR TITLE
[CARBONDATA-3322] [CARBONDATA-3323] Added check for invalid tables in ShowCacheCommand & Standard output on ShowCacheCommand on table

### PR DESCRIPTION
**Pull request by NamanRastogi**
**Problem 1**: 
After we alter table name from t1 to t2, "SHOW METACACHE ON TABLE" works for both old table name "t1" and new table name "t2".
**Fix**: 
Added check for table.

**Problem 2**: 
When "SHOW METACACHE ON TABLE" is executed and carbonLRUCAche is null, output is empty sequence, which is not standard.
**Fix**: 
Return standard output even when carbonLRUCache is not initalised (null) with size for index and dictionary as 0.

 - [x] Any interfaces changed?    ->  No
 - [x] Any backward compatibility impacted?  ->  No
 - [x] Document update required?   ->  No
 - [x] Testing done  ->  Yes
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2779/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11040/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3012/

**Comment by NamanRastogi** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2782/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11043/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3015/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 1]
```diff
@@ -74,6 +75,7 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
         .listAllTables(sparkSession).filter {
         carbonTable =>
           carbonTable.getDatabaseName.equalsIgnoreCase(currentDatabase) &&
+          isValidTable(carbonTable, sparkSession) &&
```
Comment (zzcclp) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: It's better use 'Checker.validateTableExists(databaseNameOp, tableName.get, sparkSession)'.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 2]
```diff
@@ -190,6 +192,9 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
        * Assemble result for table
        */
       val carbonTable = CarbonEnv.getCarbonTable(tableIdentifier.get)(sparkSession)
+      if (!isValidTable(carbonTable, sparkSession)) {
```
Comment (zzcclp) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: It's better use 'Checker.validateTableExists(databaseNameOp, tableName.get, sparkSession)'.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 3]
```diff
@@ -205,4 +210,9 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
       }
     }
   }
+
+  def isValidTable(carbonTable: CarbonTable, sparkSession: SparkSession): Boolean = {
```
Comment (zzcclp) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: dont need to redefine this function.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 1 above)
Comment (NamanRastogi) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: I don't need to throw an error if the table does not exist, I just need a true or false value.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 2 above)
Comment (NamanRastogi) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: Done.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 3 above)
Comment (NamanRastogi) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: `Checker.validateTableExists` throws an exception when the table is invalid, I don't need that in this case, I simply need that to be ignored.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2784/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11045/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 1 above)
Comment (zzcclp) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: Got it.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3017/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 3 above)
Comment (zzcclp) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: Got it.

**Comment by ravipesala** [comment]
LGTM, Please remove WIP.

**Comment by zzcclp** [comment]
LGTM

**Comment by brijoobopanna** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11051/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2791/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3023/

**Comment by brijoobopanna** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2797/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11057/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3029/

**Pull request closed without merging**